File: LC_Gen_Codes_CPlusPlus/2815_no_setting.cpp
Here is the converted code in Java:

java
import java.util.List;

class Solution {
    public int maxSum(List<Integer> nums) {
        int maxSum = -1;

        // Loop through each pair of numbers
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                // Find the maximum digit in both numbers
                int maxDigit = Integer.MIN_VALUE;
                String num1 = String.valueOf(nums.get(i));
                String num2 = String.valueOf(nums.get(j));

                for (char digit : num1.toCharArray()) {
                    maxDigit = Math.max(maxDigit, digit - '0');
                }
                for (char digit : num2.toCharArray()) {
                    maxDigit = Math.max(maxDigit, digit - '0');
                }

                if (maxDigit == Integer.parseInt(String.valueOf(num1.charAt(0))) || maxDigit == Integer.parseInt(String.valueOf(num2.charAt(0)))) {
                    // Calculate the sum and update maxSum if necessary
                    int pairSum = nums.get(i) + nums.get(j);
                    maxSum = Math.max(maxSum, pairSum);
                }
            }
        }

        return maxSum;
    }
}

File: LC_Gen_Codes_Java/2815_no_setting.java
class Solution {
    public int maxSum(int[] nums) {
        int max_sum = -1;
        
        // Loop through each pair of numbers
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                // Find the maximum digit in both numbers
                int max_digit = findMaxDigit(nums[i]);
                if (max_digit == findMaxDigit(nums[j])) {
                    // Calculate the sum and update max_sum if necessary
                    int pair_sum = nums[i] + nums[j];
                    max_sum = Math.max(max_sum, pair_sum);
                }
            }
        }
        
        return max_sum;
    }
    
    // Helper function to find the maximum digit in a number
    private int findMaxDigit(int num) {
        int max_digit = 0;
        while (num > 0) {
            int digit = num % 10;
            max_digit = Math.max(max_digit, digit);
            num /= 10;
        }
        return max_digit;
    }
}